huaweicloud/services/dew/resource_huaweicloud_csms_secret.go

```diff
@@ -0,0 +1,318 @@
+package dew
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/chnsz/golangsdk/openstack/common/tags"
+	"github.com/chnsz/golangsdk/openstack/csms/v1/secrets"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const (
+	serviceType = "csms"
+)
+
+func ResourceCsmsSecret() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCsmsSecretCreate,
+		ReadContext:   resourceCsmsSecretRead,
+		UpdateContext: resourceCsmsSecretUpdate,
+		DeleteContext: resourceCsmsSecretDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCsmsSecretImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[\w-\.]{1,64}$`),
+					"The maximum length is 64 characters. "+
+						"Only letters, digits, underscores (_) hyphens (-) and dots (.) are allowed."),
+			},
+			"secret_text": {
```

Comment:
Binary strings and strings are quite different, why  merged

Reply:
not merged, no support for binary data